Running B2G on Android emulator (goldfish) with ARMv7 instruction set.

ThinkerYzu opened the pull request:
This set of patches make B2G to be able running on Android emulator (goldfish) with ARMv7 instruction set.  For more detail information, please check issue #25.

ThinkerYzu committed:
Export env. ANDROID_SDK_PLATFORM and MAKE_FLAGS to gecko configuration.

We can select one of SDK platforms/versions from $ANDROID_SDK/platforms/.
MAKE_FLAGS can also be changed from cmdline or local.mk.

ThinkerYzu committed:
Add config-qemu make target for Android emulator

ThinkerYzu committed:
Switch boot/kernel-android-qemu to android-goldfish-2.6.29 branch

ThinkerYzu committed:
Build Gecko for ARMv7 instead of ARMv5

ThinkerYzu committed:
Merge branch 'master' of https://github.com/andreasgal/B2G

ThinkerYzu committed:
Remove unused module

ThinkerYzu committed:
Config kernel for Android emulator with goldfish_armv7_defconfig.

ThinkerYzu committed:
Switch Android emulator to tools_r13 branch

ThinkerYzu committed:
Build gonk with ARMv7 for QEmu

joneschrisg commented on Makefile:
```diff
@@ -177,3 +205,5 @@ sync:
 	git submodule update --init && \
 	git pull
+	@cd boot/kernel-android-qemu && \
+	git checkout origin/android-goldfish-2.6.29
```
What's the motivation for adding this to |make sync|?  What does it do that the git submodule system can't do?

ThinkerYzu replied:
These two line are no more need.  They can be removed from the patch.  Should I add another patch to revert it?

joneschrisg commented on boot/kernel-android-qemu:
```diff
@@ -1 +1 @@
-Subproject commit 29b2c49152786beae6d12988ce250b8e852c7f34
+Subproject commit 171572d0d3fd231bd4e1c71645c9e206a570a46c
```
If we're patching this kernel, we need to fork the repo on github so that we can track our changes.  Would you like to do that?  Otherwise I can tomorrow.

ThinkerYzu replied:
It is not a patch.  It switches to android-goldfish-2.6.29 branch.  It is already a part of Andreas' repository.

joneschrisg commented on glue/gonk/external/qemu:
```diff
@@ -1 +1 @@
-Subproject commit 2fa5173bd1ce7341c50adac676a317945f0969c7
+Subproject commit 14ee41d3268911e97e6fc4f417ac424336847f0a
```
Same here: we need to fork qemu on github.

ThinkerYzu replied:
This is the result of switching to tools_r23 branch.  It is already a part of Andreas' repository.

joneschrisg commented:
Again, nice work! :D

ThinkerYzu committed:
Remove useless rules.

Submodule checkouts correct branch (commit) after switching branch.
So, these two lines are useless.

joneschrisg commented:
@ThinkerYzu OK, thanks for the explanations.

This patch has "bit-rotted" and needs to be updated, but once that's done, we'll merge it in.

ThinkerYzu committed:
Merge from master
